### PR TITLE
[Components] Certifier

### PR DESCRIPTION
**Pull request by mariczne**
## WHY

Resolves https://github.com/PipedreamHQ/pipedream/issues/12426

## Summary by CodeRabbit

- **New Features**
	- Introduced "Issue Credential" action to create, issue, and send credentials using the Certifier service.
- **Enhancements**
	- Added new methods for API interactions and credential management in the Certifier app module, enhancing the overall functionality and integration capabilities.
- **Dependencies**
	- Added `@pipedream/platform` as a new dependency to support the new features and enhancements.

### CODE REVIEW

**Comment by pipedream-component-development** [comment]
Thank you so much for submitting this! We've added it to our backlog to review, and our team has been notified.

**Comment by pipedream-component-development** [comment]
Thanks for submitting this PR! When we review PRs, we follow the [Pipedream component guidelines](https://pipedream.com/docs/components/guidelines/). If you're not familiar, here's a quick checklist:
- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible

File: components/certifier/actions/issue-credential/issue-credential.mjs [hunk 1]
```diff
@@ -0,0 +1,134 @@
+import { defineAction } from "@pipedream/types";
```
Comment (michelle0927) [inline components/certifier/actions/issue-credential/issue-credential.mjs:1]:
```suggestion
```

File: components/certifier/actions/issue-credential/issue-credential.mjs [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+import certifier from "../../certifier.app.mjs";
+
+export default defineAction({
```
Comment (michelle0927) [inline components/certifier/actions/issue-credential/issue-credential.mjs:4]:
```suggestion
export default {
```

File: components/certifier/actions/issue-credential/issue-credential.mjs [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+  name: "Issue Credential",
+  version: "0.0.1",
+  key: "certifier-issue-credential",
+  description: "Create, issue and send a credential to a recipient.",
```
Comment (michelle0927) [inline components/certifier/actions/issue-credential/issue-credential.mjs:8]:
```suggestion
  description: "Create, issue and send a credential to a recipient. [See the documentation](https://developers.certifier.io/reference/issue-a-credential)",
```

File: components/certifier/actions/issue-credential/issue-credential.mjs [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+    const customAttributes = Object.fromEntries(
+      Object.entries(this).filter(([
+        key,
+      ]) => key.startsWith("custom.")),
+    );
+
+    let response = await certifier.createCredential({
+      data: {
+        groupId: groupId,
+        recipient: {
```
Comment (michelle0927) [inline components/certifier/actions/issue-credential/issue-credential.mjs:97]:
```suggestion
    let response = await certifier.createCredential({
      $,
      data: {
        groupId: groupId,
        recipient: {
```

File: components/certifier/actions/issue-credential/issue-credential.mjs [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+        ...(expiryDate && {
+          expiryDate,
+        }),
+      },
+    });
+
+    console.log(`Successfully created credential with ID \`${response.id}\`.`);
+
+    if (issueCredential) {
+      response = await certifier.issueCredential(response.id);
```
Comment (michelle0927) [inline components/certifier/actions/issue-credential/issue-credential.mjs:114]:
```suggestion
      response = await certifier.issueCredential(response.id, { $ });
```

File: components/certifier/actions/issue-credential/issue-credential.mjs [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+
+      console.log(`Successfully issued credential with ID \`${response.id}\`.`);
+
+      if (sendCredential) {
+        response = await certifier.sendCredential(response.id, {
+          data: {
+            deliveryMethod: "email",
+          },
+        });
```
Comment (michelle0927) [inline components/certifier/actions/issue-credential/issue-credential.mjs:123]:
```suggestion
        response = await certifier.sendCredential(response.id, {
          $,
          data: {
            deliveryMethod: "email",
          },
        });
```

File: components/certifier/actions/issue-credential/issue-credential.mjs [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+        console.log(`Successfully sent credential with ID \`${response.id}\`.`);
+      }
+    }
+
+    $.export("$summary",
+      `Successfully created credential for ${response.recipient.name}`);
+
+    return response;
+  },
+});
```
Comment (michelle0927) [inline components/certifier/actions/issue-credential/issue-credential.mjs:134]:
```suggestion
};
```

File: components/certifier/certifier.app.mjs [hunk 8]
```diff
@@ -1,11 +1,126 @@
-export default {
+import { axios } from "@pipedream/platform";
+import { defineApp } from "@pipedream/types";
```
Comment (michelle0927) [inline components/certifier/certifier.app.mjs:2]:
```suggestion
```

File: components/certifier/certifier.app.mjs [hunk 9]
```diff
@@ -1,11 +1,126 @@
...
+
+export default defineApp({
```
Comment (michelle0927) [inline components/certifier/certifier.app.mjs:4]:
```suggestion
export default {
```

File: components/certifier/certifier.app.mjs [hunk 10]
```diff
@@ -1,11 +1,126 @@
...
+      type: "string",
+      label: "Expiry Date",
+      description:
+        "The date (in `YYYY-MM-DD` format) of your credential's expiration. If not provided, expiry date from the group settings will be used (by default this field is empty).",
+      optional: true,
     },
   },
-};
\ No newline at end of file
+});
```
Comment (michelle0927) [inline components/certifier/certifier.app.mjs:126]:
```suggestion
};
```

File: components/certifier/package.json [hunk 11]
```diff
@@ -12,4 +12,4 @@
   "publishConfig": {
     "access": "public"
   }
-}
\ No newline at end of file
+}
```
Comment (michelle0927) [inline components/certifier/package.json:15]: Need to add `@pipedream/platform` dependency.
```
  "dependencies": {
    "@pipedream/platform": "^3.0.0"
  }
```

**Review by michelle0927 — CHANGES_REQUESTED** [review]
Thank you so much for your contribution! I left some comments about some small changes, such as including the `$` variable. I tried to make the changes myself, but was denied permission for some reason. Let me know if you have any questions or need any help with the updates.

File: components/certifier/actions/issue-credential/issue-credential.mjs (hunk 3 above)
Comment (mariczne) [reply components/certifier/actions/issue-credential/issue-credential.mjs:8]: This is probably not the best page we could link to. I see how the naming could be confusing 😞 I'll change the link to https://api.certifier.io/v1/credentials/create-issue-send, which is a similar multi-action.

**Comment by mariczne** [comment]
Thank you for the review @michelle0927! I've incorporated the changes you requested, with one small difference mentioned in another comment.

**Review by lcaresia — APPROVED** [review]

**Review by michelle0927 — APPROVED** [review]

**Pull request merged**